contracts/src/FunctionGatewayV2.sol

```diff
@@ -0,0 +1,165 @@
+contract FunctionGatewayUpdated {
+    mapping(bytes32 => address) functionIdToVerifier;
+
+    function callbackRequestV1(
+        bytes32 _functionId,
```

Comment:
I feel like a better name for this function is something like `requestWithContext`

Reply:
and likewise `fulfillRequestWithContextV1`